src/librustdoc/html/format.rs

```diff
@@ -456,15 +456,26 @@ impl clean::GenericArgs {
     }
 }
 
-crate fn href(did: DefId, cx: &Context<'_>) -> Option<(String, ItemType, Vec<String>)> {
+// Possible errors when computing href link source for a `DefId`
+crate enum HrefError {
+    // `DefId` is in an unknown location. This seems to happen when building without dependencies
+    // but a trait from a dependency is still visible
+    UnknownLocation,
```

Comment:
This happens when `cache.extern_locations[&did.krate]` is `ExternLocation::Unknown`. That has nothing to do with traits, it can happen for any item from a crate without documentation built.
```suggestion
    /// This item is known to rustdoc, but from a crate that does not have documentation generated.
    ///
    /// This can only happen for non-local items.
    DocumentationNotBuilt,
```